Simulation/Derivatives.m

```diff
@@ -1,17 +1,28 @@
-function StateVector_dot = Derivatives(StateVector, Object)
-  Velocity = StateVector(3:4);
+function kOut = Derivatives(kn, SimData)
+    % Add k Deltas to the current SimData
+    SimData.StateVector.Position    = SimData.StateVector.Position + kn.P_delta;
+    SimData.StateVector.Velocity    = SimData.StateVector.Velocity + kn.V_delta;
+    SimData.StateVector.Orientation = SimData.StateVector.Orientation + kn.O_delta;
 
-  Gravity = [0, -9.81]; % Acceleration due to gravity
+    % Acceleration due to gravity
```

Comment:
Formatting is a bit off here because we use 2 spaces everywhere instead of 4 here.

Reply:
Fixed